pkg/codegen/schema/bind.go

```diff
@@ -958,35 +958,46 @@ func bindConstValue(path, kind string, value interface{}, typ Type) (interface{}
 
 	switch typ = plainType(typ); typ {
 	case BoolType:
-		if _, ok := value.(bool); !ok {
+		v, ok := value.(bool)
+		if !ok {
 			return false, typeError("boolean")
 		}
+		return v, nil
 	case IntType:
-		v, ok := value.(float64)
+		v, ok := value.(int)
 		if !ok {
-			return 0, typeError("integer")
+			v, ok := value.(float64)
+			if !ok {
+				return 0, typeError("integer")
+			}
+			if math.Trunc(v) != v || v < math.MinInt32 || v > math.MaxInt32 {
+				return 0, typeError("integer")
+			}
+			return int32(v), nil
 		}
-		if math.Trunc(v) != v || v < math.MinInt32 || v > math.MaxInt32 {
+		if v < math.MinInt32 || v > math.MaxInt32 {
```

Comment:
Is this check ever needed when `v` was correctly casted to `int` from `value`?

Reply:
Good question. That check takes a float64.

Reply:
Yeah. I've written a test case to catch overflows/underflows that failed without it.